src/particle.cpp

```diff
@@ -233,13 +234,15 @@ void Particle::event_advance()
     coord(j).r += distance * coord(j).u;
   }
   this->time() += distance / this->speed();
+  this->lifetime() += distance / this->speed();
```

Comment:
Given that `lifetime` looks identical to `time` here, can you explain what the difference between the two is?

Reply:
The main difference between time() and lifetime() is that when a particle is created via Particle::from_source, lifetime() will be reset to 0 while time() will be set to the source time. This means that time() will not necessarily be equal to 0 when Particle::from_source is called if we already did multiple batches. In my case, I really need a timer that is reset every time a particle is created.